src/ray/raylet/scheduling/cluster_resource_scheduler.cc

```diff
@@ -262,10 +262,14 @@ int64_t ClusterResourceScheduler::GetBestSchedulableNode(
     *total_violations = 0;
   }
 
+  std::string best_node_id_str = "NA";
+  if (best_node_id != -1) {
+    best_node_id_str = NodeID::FromBinary(string_to_int_map_.Get(best_node_id)).Hex();
+  }
+
   RAY_LOG(DEBUG) << "Scheduling decision. "
                  << "forcing spillback: " << force_spillback
-                 << ". Best node: " << best_node_id << " "
-                 << NodeID::FromBinary(string_to_int_map_.Get(best_node_id))
```

Comment:
Thanks for catching this! Could we do in-line ?: so that it's only run when debug is on? Also you don't need hex().

Reply:
Done! And Hex() is needed because there is no conversion method from NodeID to string.

Reply:
I think we can do
```
(best_node_id == -1
                         ? NodeID::Nil()
                         : NodeID::FromBinary(string_to_int_map_.Get(best_node_id)))
```

Reply:
Nil will print FFFFFFFFFFFFFFFF,  I think NA is easier to understand?

Reply:
Or just an empty string

Reply:
I feel like then the fix should be in NodeID << operator so it prints a more understandable string for Nil case, this way we get the same  benefit wherever we print NodeID? Personally I feel like calling hex() directly is relying on the << implementation, which can be changed.

Reply:
I think FFFFF is not bad since it is not user-facing API (my assumption is most of core devs know fffff is the nil id?)

Reply:
Yeah I agree with you. I changed it to Nil(), please have a look :)